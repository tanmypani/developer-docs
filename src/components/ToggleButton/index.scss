@import '../../assets/styles/variables.scss';

.switch {
    position: relative;
    display: inline-block;
    width: $switch-width;
    height: $switch-height;
    padding-left: $padding-md;
}

.switch input {
    display: none;
}

.slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: $slider-background;
    -webkit-transition: 0.4s;
    transition: 0.4s;
    border-radius: 34px;
}

.slider:before {
    position: absolute;
    content: '';
    height: $slider-width;
    width: $slider-height;
    left: 4px;
    bottom: 4px;
    background-color: white;
    -webkit-transition: 0.4s;
    transition: 0.4s;
    border-radius: 50%;
}

input:checked + .slider {
    background-color: $gray-80;
}

input:checked + .slider:before {
    -webkit-transform: translateX(26px);
    -ms-transform: translateX(26px);
    transform: translateX(45px);
}

.slider:after {
    content: 'Light';
    color: var(--primary-color);
    display: block;
    position: absolute;
    transform: translate(-50%, -50%);
    top: 50%;
    right: 0;
    font-size: 12px;
    font-family: Verdana, sans-serif;
}

input:checked + .slider:after {
    content: 'Dark';
    color: var(--primary-color);
    left: 40%;
}

