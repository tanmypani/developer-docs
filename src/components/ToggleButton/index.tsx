import React, { useEffect } from 'react';
import './index.scss';

const ToggleButton = (props: {
    setDarkMode: Function;
    isDarkMode: boolean;
}) => {
    return (
        <label className="switch">
            <input
                data-testid="toggle-btn"
                type="checkbox"
                id="togBtn"
                onChange={(e) => {
                    props.setDarkMode(e.target.checked);
                    localStorage.setItem(
                        'theme',
                        e.target.checked ? 'dark' : 'light',
                    );
                }}
                checked={props.isDarkMode}
            />
            <div data-testid="slider" className="slider round"></div>
        </label>
    );
};

export default ToggleButton;
