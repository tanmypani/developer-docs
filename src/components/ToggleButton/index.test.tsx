import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { toHaveStyle } from '@testing-library/jest-dom'

import ToggleButton from './index';

describe('ToggleButton', () => {

    const setDarkMode = jest.fn();
    const isDarkMode = false;
    it("should renders correctly", () => {
        const { container } = render(<ToggleButton setDarkMode={setDarkMode} isDarkMode={isDarkMode} />);

        expect(container).toMatchSnapshot()
    })

    it("should toggle correctly", () => {
        const { getByTestId, getByText } = render(<ToggleButton setDarkMode={setDarkMode} isDarkMode={isDarkMode} />);
        const toggleBtn = getByTestId('toggle-btn');
        act(() => {
            fireEvent.click(toggleBtn)
        })
        expect(setDarkMode).toHaveBeenCalledTimes(1);
        expect(setDarkMode).toHaveBeenCalledWith(true);
    })
})