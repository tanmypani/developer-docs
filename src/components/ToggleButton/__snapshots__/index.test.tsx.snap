// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton should renders correctly 1`] = `
<div>
  <label
    class="switch"
  >
    <input
      data-testid="toggle-btn"
      id="togBtn"
      type="checkbox"
    />
    <div
      class="slider round"
      data-testid="slider"
    />
  </label>
</div>
`;
